### PR TITLE
Make Remark an optional field

### LINKED ISSUE #35: Make Remark an optional field

### PR DESCRIPTION
**Pull request by dorcastan**
Fixes #35 

Changes:

1. Not necessary to include a remark when adding bookmarks.
2. Empty remarks (i.e. `r/` and `r/   `) will be replaced by a default empty value (`-`).
3. Remarks containing `/` are invalid.

Miscellaneous fixes:

1. Bookmarks are recognised as identical if *either* name or URL is the same.
2. Broken link to `Command` class in `CommandException` javadoc

### CODE REVIEW

File: src/main/java/seedu/mark/logic/commands/exceptions/CommandException.java [hunk 1]
```diff
@@ -1,5 +1,7 @@
 package seedu.mark.logic.commands.exceptions;
 
+import seedu.mark.logic.commands.Command;
```
Comment (Na-Nazhou) [inline src/main/java/seedu/mark/logic/commands/exceptions/CommandException.java]: Why is this line added? I did not find any usage of `Command`

File: src/main/java/seedu/mark/model/bookmark/Remark.java [hunk 2]
```diff
@@ -9,14 +9,17 @@
...
 public class Remark {
 
-    public static final String MESSAGE_CONSTRAINTS = "Remarks can take any values, and it should not be blank";
+    public static final String SPECIAL_CHARACTER = "/";
+
+    public static final String MESSAGE_CONSTRAINTS = "Remarks can contain any characters except " + SPECIAL_CHARACTER;
 
     /*
      * The first character of the remark must not be a whitespace,
      * otherwise " " (a blank string) becomes a valid input.
```
Comment (Na-Nazhou) [inline src/main/java/seedu/mark/model/bookmark/Remark.java]: You can mention the special character in the javadoc comment as well

Comment (dorcastan) [reply src/main/java/seedu/mark/model/bookmark/Remark.java]: I left it out as it's written in `MESSAGE_CONSTRAINTS`, directly above this. Should also include it here?

Comment (openorclose) [reply src/main/java/seedu/mark/model/bookmark/Remark.java]: Can we rename it to INVALID_CHARACTER instead?

Comment (dorcastan) [reply src/main/java/seedu/mark/model/bookmark/Remark.java]: Done! Okay to merge?
